Classes/Http/DetectContentSubgraphComponent.php

```diff
@@ -15,15 +15,21 @@
 use Neos\ContentRepository\Domain\Service\ContextFactoryInterface;
 use Neos\ContentRepository\Domain\Utility\NodePaths;
 use Neos\Flow\Annotations as Flow;
-use Neos\Flow\Http;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
 use Neos\Flow\Mvc\Routing\Dto\RouteParameters;
-use Neos\Flow\Mvc\Routing\RoutingComponent;
+use Neos\Flow\Http\ServerRequestAttributes;
+
 use Flowpack\Neos\DimensionResolver\Http\ContentDimensionDetection\DimensionPresetDetectorResolver;
 
 /**
  * The HTTP component for detecting the requested dimension space point
  */
-final class DetectContentSubgraphComponent implements Http\Component\ComponentInterface
+final class DetectContentSubgraphComponent implements MiddlewareInterface
```

Comment:
This should probably be renamed to `DetectContentSubgraphMiddleware` to avoid confusion!?